### ISSUE TITLE
Twin problems with the bpz_lite stage datapath variable

### ISSUE THREAD

**Issue by joezuntz** [LSSTDESC/RAIL#266]
Hi all.  Another problem I've seen while trying to run CosmoDC2 through a pipeline on NERSC.

One of the limitations of running under shifter is that the file system containing the python libraries is read-only. That makes the default BPZ path fail with this error:

```
  ...
  File "/opt/conda/lib/python3.9/site-packages/desc_bpz/bpz_tools_py3.py", line 749, in ABflux
    put_data(ABoutput,(z_ab,f))
  File "/opt/conda/lib/python3.9/site-packages/desc_bpz/useful_py3.py", line 151, in write
    else: f=open(file,'w')
OSError: [Errno 30] Read-only file system: '/opt/conda/lib/python3.9/site-packages/rail/examples/estimation/data/AB/El_B2004a.DC2LSST_u.AB'
```

One possible fix would be to pre-create these files in the package, I guess. Or I could do this as part of the docker build, that would work too.

This looks like exactly what the `BPZDATAPATH` environment variable is supposed to fix, so I tried setting this via the data_path configuration option. There is a issue with that in ceci which I will open an issue about on that repo.

**Comment by sschmidt23** [comment]
Yes, if not present these files are created automatically by DESC_BPZ, however that obviously won't work on a read-only filesystem.  As you say, you can pre-create them (by running rail_bpz or DESC_BPZ with even one fake or real galaxy beforehand on a system that is not read-only) and then specify the path to the AB directory via BPZDATAPATH.  For a stable set of SEDs and FILTERs that was the assumed procedure.  You can also copy the SED/FILTER files to a location outside of the python library directory structure and set the BPZDATAPATH to that location, which presumably is not read-only, i.e. just copy the AB/SED/FILTER directories in `src/rail/examples/estimation/data` to a path on cori that you can write to.

We could include the AB files for the default filters in the rail_bpz repo, but if you're using anything other than the default SED + default FILTER combo you would still have the same problem.

**Comment by joezuntz** [comment]
Thanks - that all makes sense. With Eric's fix to the parameter setting issue I opened on ceci this is solvable now.  I guess I'll just include the basic files in the input files directory.

One other thing I spotted, though, is that if you're running in parallel then the current design can cause a crash, if you're unlucky - all the processes can try to generate the AB files at the same time. I'll put in a quick PR for that. Also I think there's no reason to be re-loading the templates for every chunk? I'll fix that too.

**Comment by OliviaLynn** [comment]
Alex said this would be a good place to mention a path problem I’ve run into with bpz_rail — bpz assumes itself to be stored in a certain place in the user’s RAIL directory structure, and it uses the root of the RAIL directory to navigate its own directory system. 

For example, the bpz columns file path is set on line 256 of `bpz_lite.py` to be `os.path.join(RAILDIR, 'rail/examples/estimation/configs/test_bpz.columns')`, which probably works great if you have them both installed via default settings — but it introduces problems if you’ve installed RAIL outside of `site-packages/` for development purposes, or whatever reason.

For now I have my install working by rewriting path references where they don’t make sense, but this could be worth keeping an eye on or looking into in the future.
